### ISSUE TITLE
Move `build` from positional args to named.

### ISSUE THREAD

**Issue by supakeen** [osbuild/image-builder-cli#31]
It'd be nicer if the `image-builder build` subcommand takes named arguments instead of positional arguments. Currently it looks like this:

```
image-builder build --distro fedora-41 minimal-raw blueprint.toml
```

I propose:

```
image-builder build --distro fedora-41 --type minimal-raw --blueprint blueprint.toml
```

This would allow for multi-type builds (a very strong advantage of the `osbuild` system):

```
image-builder build --distro fedora-41 --type iso --type qcow2 --blueprint blueprint.toml
```

And would allow us to in the future have other inputs except the built-in image types such as a custom YAML file for custom images.

**Comment by mvo5** [comment]
Fwiw, I like that it's more "uniform" this way, i.e. to me it feels a bit unbalanced that we currently have "--distro centos-9" but type is just an positional argument and has no --type. But then its a bit more typing for our users. 

We should probably have a brainstorm again in the new year.

**Comment by achilleas-k** [comment]
I agree, there's a convenience to positional args, but I think once you go beyond 3, it starts becoming easier to forget which is which.  I think the extra typing is an acceptable tradeoff for explicitness.  I wouldn't be against going that way.

**Comment by mvo5** [comment]
@ondrejbudai Any (strong) opinions here?

**Comment by ondrejbudai** [comment]
I only have one strong opinion: the user experience for the first time users must be as simple as possible. I like that positional arguments is all you need to get an image, I find it extremely user-friendly: `image-builder build qcow2`.

Requiring a named argument makes the command longer, and imo slightly harder to document.

---

Would it make sense to split these use-cases into 2 subcommands? One very simple one (`build`) for the masses, one complex one for people that need the complexity to succeed. I'm even thinking that the complex command shouldn't accept arguments, but rather a file (might be an overkill?):

```console
$ cat fedora-iot.yaml
ostree-url: example.com/fedora
ostree-ref: fedora/rawhide
types:
  - iot-installer
  - iot-simplified-installer
  - iot-raw
  - iot-container
distro: fedora-rawhide
$ image-builder-cli multi-build fedora-iot.yaml
```

**Comment by supakeen** [comment]
I'm fine with either approach; I created a quick PR to have `--type` because it was very little work to have it.

I'll add an opinion that if we do what @ondrejbudai proposes to please call these commands `build-multi` (so the reverse) so they sort correctly (all under `build` :)).

**Comment by supakeen** [comment]
We've settled on having `--blueprint`; which was merged in #65. So let's close this one.
